src/helpers/nftLinksFactory.js

```diff
@@ -0,0 +1,74 @@
+import IconOpensea from "~/renderer/icons/Opensea";
+import IconRarible from "~/renderer/icons/Rarible";
+import IconGlobe from "~/renderer/icons/Globe";
+import { openURL } from "~/renderer/linking";
+
+const linksPerCurrency = {
+  ethereum: (t, links) => [
+    links?.opensea && {
+      key: "opensea",
+      id: "opensea",
+      label: t("NFT.viewer.actions.open", { viewer: "Opensea.io" }),
+      Icon: IconOpensea,
+      type: "external",
+      callback: () => openURL(links.opensea),
+    },
+    links?.rarible && {
+      key: "rarible",
+      id: "rarible",
+      label: t("NFT.viewer.actions.open", { viewer: "Rarible" }),
+      Icon: IconRarible,
+      type: "external",
+      callback: () => openURL(links.rarible),
+    },
+    {
+      key: "sep2",
+      id: "sep2",
+      type: "separator",
+      label: "",
+    },
+    links?.etherscan && {
+      key: "etherscan",
+      id: "etherscan",
+      label: t("NFT.viewer.actions.open", { viewer: "Explorer" }),
+      Icon: IconGlobe,
+      type: "external",
+      callback: () => openURL(links.etherscan),
+    },
+  ],
+  polygon: (t, links) => [
+    links?.opensea && {
+      key: "opensea",
+      id: "opensea",
+      label: t("NFT.viewer.actions.open", { viewer: "Opensea.io" }),
+      Icon: IconOpensea,
+      type: "external",
+      callback: () => openURL(links.opensea),
+    },
+    links?.rarible && {
+      key: "rarible",
+      id: "rarible",
+      label: t("NFT.viewer.actions.open", { viewer: "Rarible" }),
+      Icon: IconRarible,
+      type: "external",
+      callback: () => openURL(links.rarible),
+    },
+    {
+      key: "sep2",
+      id: "sep2",
+      type: "separator",
+      label: "",
+    },
+    links?.polygonscan && {
+      key: "polygonscan",
```

Comment:
dumb question but, if the key is always the id, do we need both?

Reply:
We're using the "key" and "id" at different places in our code, that's why I added both :( 
`DropDownItemType` uses `key`: 
`ContextMenuItemType` uses `id`: https://github.com/LedgerHQ/ledger-live-desktop/blob/adee1a8f35fa7ebdc238745e8109660fbe759514/src/renderer/components/ContextMenu/ContextMenuWrapper.js#L27

Changing every ContextMenu or every DropDown just for that felt too risky for no rewards 😞

Reply:
aight aight, dumb question